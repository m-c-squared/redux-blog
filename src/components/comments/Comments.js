import React from 'react';
import PropTypes from 'prop-types';
import styles from 'styled-components';
import { HomePosts } from '../posts/LongestPosts';

export const CommentsList = styles.ul`
@media (min-width: 374px) {
  display: grid:
  grid-template-columns: 33% 33% 33%; 
  padding: 0;
  }
  @media (min-width: 600px) {
    display: grid;
    grid-template-columns: 33% 33% 33%; 
    padding: 0;
    }
`;

export const CommentsHeader = styles.h3`
@media (min-width: 374px) {
  text-align: left;
  font-family: 'Roboto', san-serif;
  margin: 2em .5em 2em .5em;
  color: rgb(93, 95, 96);
}
`;



const Comments = ({ comments }) => {

  const commentList = comments.map((({ body }) => {
    return (<CommentsHeader key={body}>Comment<HomePosts> {body} </HomePosts></CommentsHeader>) ;
  }));

  return (
    <>
    <CommentsList>  
      {commentList}
    </CommentsList>
     </>
  );
};

Comments.propTypes =  {
  comments: PropTypes.array.isRequired
};

export default Comments; 
