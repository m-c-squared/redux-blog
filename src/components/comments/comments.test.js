import React from 'react';
import renderer from 'react-test-renderer';
import Comments from './Comments';
import { MemoryRouter } from 'react-router-dom';


describe('Comments', () => {
  it('matches a snapshot', () => {
    const comments = [
      {
        body: 'pariatur occaecati ea autem at quis et dolorem similique pariatur ipsa hic et saepe itaque cumque repellendus vel et quibusdam qui aut nemo et illo qui non quod officiis aspernatur qui optio'
      },
      {
        body: 'et dolorem sit reprehenderit sapiente occaecati iusto sit impedit nobis ut quia maiores debitis pariatur nostrum et aut assumenda error qui deserunt laborum quaerat et'  
      }
    ];
    const tree = renderer.create(
      <MemoryRouter>
        <Comments comments={comments}/>
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
