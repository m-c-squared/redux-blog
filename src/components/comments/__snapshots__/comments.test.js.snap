// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments matches a snapshot 1`] = `
<ul
  className="sc-htpNat hqwPgn"
>
  <h3
    className="sc-bxivhb blXckI"
  >
    Comment
    <li
      className="sc-bwzfXH fqQUGx"
    >
       
      pariatur occaecati ea autem at quis et dolorem similique pariatur ipsa hic et saepe itaque cumque repellendus vel et quibusdam qui aut nemo et illo qui non quod officiis aspernatur qui optio
       
    </li>
  </h3>
  <h3
    className="sc-bxivhb blXckI"
  >
    Comment
    <li
      className="sc-bwzfXH fqQUGx"
    >
       
      et dolorem sit reprehenderit sapiente occaecati iusto sit impedit nobis ut quia maiores debitis pariatur nostrum et aut assumenda error qui deserunt laborum quaerat et
       
    </li>
  </h3>
</ul>
`;
